### PR TITLE
Add support for module local preview.

### PR DESCRIPTION
**Pull request by cube2222**
This is mostly a copy-paste of stack local-preview, with the addition of some fancy status rendering.

### CODE REVIEW

**Comment by Clickup-user** [comment]
Task linked: [CU-2d2qx25 Local preview for module tests](https://app.clickup.com/t/2d2qx25)

File: internal/local_preview.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		return fmt.Errorf("couldn't create upload request: %w", err)
+	}
+	req.ContentLength = stat.Size()
+	req = req.WithContext(ctx)
+
+	response, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("couldn't upload workspace: %w", err)
+	}
+	defer response.Body.Close()
```
[the remark this answers is not in the document]
Comment (cube2222) [reply internal/local_preview.go:101]: This is fine.

**Review by adamconnelly — APPROVED** [review]
Looks neat. Have you got a screenshot you can share?

File: internal/cmd/module/local_preview.go [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+						newRuns[index] = getRun.Module.Run
+
+						return nil
+					})
+				}
+				if err := g.Wait(); err != nil {
+					log.Fatal("couldn't get runs: ", err)
+				}
+				model.Lock()
```
[the remark this answers is not in the document]
Comment (cube2222) [reply internal/cmd/module/local_preview.go:131]: Good point, will do.

**Pull request merged**